rosidl_generator_py/resource/_msg_support.c.em

```diff
@@ -344,14 +344,15 @@ nested_type = '__'.join(type_.namespaced_name())
       char * buffer;
       Py_ssize_t length;
       int rc = PyBytes_AsStringAndSize(encoded_item, &buffer, &length);
-      Py_DECREF(encoded_item);
       if (rc) {
+        Py_DECREF(encoded_item);
         Py_DECREF(seq_field);
         Py_DECREF(field);
         return false;
       }
       // use offset of 2 to skip BOM mark
       bool succeeded = rosidl_runtime_c__U16String__assignn_from_char(&dest[i], buffer + 2, length - 2);
```

Comment:
if I understand `PyBytes_AsStringAndSize` documentation correctly, we had an use after free problem, right?

Reply:
Correct. As of Python 3.8 the memory values have changed a bit:

> Byte patterns `0xCB` (`CLEANBYTE`), `0xDB` (`DEADBYTE`) and `0xFB` (`FORBIDDENBYTE`) have been replaced with `0xCD`, `0xDD` and `0xFD` to use the same values than Windows CRT debug `malloc()` and `free()`.